create-mc-app fails during npm install

**Describe the issue**

Installing a merchant center application template via `npx @commercetools-frontend/create-mc-app@latest my-new-custom-application-project --template starter` fails with the following error message during the `npm install` step:
```
...
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: my-new-custom-application-project@1.0.0
npm ERR! Found: graphql@16.6.0
npm ERR! node_modules/graphql
npm ERR!   graphql@"^16.6.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer graphql@"^0.12.0 || ^0.13.0 || ^14.0.0 || ^15.0.0" from eslint-plugin-graphql@4.0.0
npm ERR! node_modules/eslint-plugin-graphql
npm ERR!   eslint-plugin-graphql@"^4.0.0" from the root project
...
```
After running `create-mc-app`, the template project is created in the active directory. 
However, Running `npm start` the template directory results in an error similar to the one above.

**To Reproduce**
Steps to reproduce the behavior:

1. Open terminal in empty directory
2. Enter the following: 
```
npx @commercetools-frontend/create-mc-app@latest \
  my-new-custom-application-project \
  --template starter
```
3. Observe error in console output

**Expected behavior**

`create-mc-app` downloads and builds the template application with no error. 

**Screenshots**

<img width="567" alt="Screen Shot 2023-06-09 at 10 48 30 AM" src="https://github.com/commercetools/merchant-center-application-kit/assets/135636519/5a62e78c-aab4-4033-8b48-7188de763fe4">

**Additional context**
Logs from a failed run of `create-mc-app`:
[2023-06-09T14_46_52_505Z-debug-0.log](https://github.com/commercetools/merchant-center-application-kit/files/11707423/2023-06-09T14_46_52_505Z-debug-0.log)
[2023-06-09T14_46_52_505Z-eresolve-report.txt](https://github.com/commercetools/merchant-center-application-kit/files/11707424/2023-06-09T14_46_52_505Z-eresolve-report.txt)

Fails for both `--template=starter` and `--template=starter-typescript`
System info:
- macOS Monterey `12.6.6`
- Node `20.2.0` and `18.13.0` (tried on both)
- npm `9.6.6` and `8.19.3` (tried on both)

Initial debugging included changing `graphql` in `package.json` to `^15.0.0` and while that fixed the initial dependency conflict, more conflicts presented themself.

Template was able to be successfully built and ran via `npm install --legacy-peer-deps` and `npm start`.

--- a/.changeset/fresh-lizards-remain.md
+++ b/.changeset/fresh-lizards-remain.md
@@ -1,0 +1,5 @@
+---
+'@commercetools-frontend/create-mc-app': patch
+---
+
+Run `npm install` command with `--legacy-peer-deps` option

--- a/packages/create-mc-app/src/tasks/install-dependencies.ts
+++ b/packages/create-mc-app/src/tasks/install-dependencies.ts
@@ -1,17 +1,17 @@
 import execa from 'execa';
 import type { ListrTask } from 'listr2';
 import type { TCliTaskOptions } from '../types';
-import { getPreferredPackageManager } from '../utils';
+import { getInstallCommand } from '../utils';
 
 function installDependencies(options: TCliTaskOptions): ListrTask {
   return {
     title: 'Installing dependencies (this might take a while)',
     task: () => {
-      const packageManager = getPreferredPackageManager(options);
+      const installCommand = getInstallCommand(options);
 
       // TODO: we could check for min yarn/npm versions
       // See https://github.com/facebook/create-react-app/blob/0f4781e8507249ce29a9ac1409fece67c1a53c38/packages/create-react-app/createReactApp.js#L225-L254
-      return execa(packageManager, ['install'], {
+      return execa.command(installCommand, {
         cwd: options.projectDirectoryPath,
         encoding: 'utf-8',
       });

--- a/packages/create-mc-app/src/utils.ts
+++ b/packages/create-mc-app/src/utils.ts
@@ -27,6 +27,17 @@ const getPreferredPackageManager = (
   return 'npm';
 };
 
+const getInstallCommand = (options: TCliTaskOptions): string => {
+  const packageManager = getPreferredPackageManager(options);
+
+  switch (packageManager) {
+    case 'npm':
+      return 'npm install --legacy-peer-deps';
+    default:
+      return `${packageManager} install`;
+  }
+};
+
 const slugify = (name: string) => name.toLowerCase().replace(/_/gi, '-');
 
 const upperFirst = (value: string) =>
@@ -54,4 +65,5 @@ export {
   upperFirst,
   resolveFilePathByExtension,
   getPreferredPackageManager,
+  getInstallCommand,
 };

